File: collector/src/main/java/com/bitmovin/analytics/data/LicenseResponse.java
package com.bitmovin.analytics.data;

import androidx.annotation.Keep;

import com.bitmovin.analytics.license.FeatureConfigContainer;

@Keep // Protect from obfuscation in case customers are using proguard
public class LicenseResponse {
    private String status;
    private String message;
    private FeatureConfigContainer features;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public FeatureConfigContainer getFeatures() {
        return features;
    }

    public void setFeatures(FeatureConfigContainer features) {
        this.features = features;
    }
}

File: collector/src/main/java/com/bitmovin/analytics/data/LicenseResponse.kt
package com.bitmovin.analytics.data

import com.bitmovin.analytics.license.FeatureConfigContainer

// DTO which is protected from ProGuard obfuscation through proguard-consumer-rules.pro
data class LicenseResponse(val status: String?, val message: String?, val features: FeatureConfigContainer?)
